pkg/serviceaccount/claims.go

```diff
@@ -166,6 +187,18 @@ func (v *validator) Validate(_ string, public *jwt.Claims, privateObj interface{
 		podUID = podref.UID
 	}
 
+	// Check special 'warnafter' field for projected service account token transition.
+	warnafter := private.Kubernetes.WarnAfter
+	if warnafter != 0 {
+		if now.After(warnafter.Time()) {
+			// TODO: add this information into audit event after #89305 got merged
+			klog.Infof("Stale service account token used, service account: %s", public.Subject)
```

Comment:
This should be logged at WARN level

Reply:
This is expected. After #89305, we should remove logging altogether.